Bdd unit test

bektasisik opened the pull request:
controller ve service katmanlarının bdd classları yazıldı

bektasisik committed:
birden fazla get request işlminde tek @GetMapping kullanarak olayı handle ettim

bektasisik committed:
BDD test katmanı eklendi

yasinbuzgulu reviewed (changes requested):
Testleri için ilerleme yanlış olmuş, Emre abi söylemiş sanırım nasıl olması gerektiğini bu epey yanlış bir ilerleme onun için Emre abinin söylediği değişikleri yapınca tekrar pr atarsın.

bektasisik committed:
Bir adet test senaryosu eklendi
//todo:
run edildi ama Test Results 'da scneriolar gözükmüyor

bektasisik committed:
Bir adet test senaryosu eklendi
//todo:
run edildi ama Test Results 'da scneriolar gözükmüyor
çözüldü

bektasisik committed:
StudentCrud işlemleri için test senaryoları eklendi

bektasisik committed:
Attendance.feature ve StudentAttendance.feature dosyaları içine senaryolar yazıldı
Step definitionları implement edildi

bektasisik committed:
AttendanceService dosyasında stream.Collectors.toList() hatasını çözdük
artık yoklama güncelleme işlemi sorunsuzca çözülüyor
feature ve stepDefinitions dosyalarındaki aldıgım hataları çözdüm
